nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -164,7 +165,15 @@ def init_model(self, cfg: DictConfig, trainer: Trainer):
 
         # Prepare pseudo token ids for virtual/virtual prompt tokens
         self.pseudo_tokens = get_pseudo_tokens(self.max_virtual_tokens)
-        self.tokenizer.add_special_tokens({'additional_special_tokens': self.pseudo_tokens})
+        if isinstance(self.tokenizer, SentencePieceTokenizer):
+            if not self.tokenizer.legacy:
+                self.tokenizer.pad_token = self.tokenizer.ids_to_tokens([self.tokenizer.pad_id])[0]
+                self.tokenizer.bos_token = self.tokenizer.ids_to_tokens([self.tokenizer.bos_id])[0]
+                self.tokenizer.eos_token = self.tokenizer.ids_to_tokens([self.tokenizer.eos_id])[0]
+                self.tokenizer.legacy = True
```

Comment:
I haven't worked with SentencePiece tokenizer so it's best to check with @MaximumEntropy 

One thing that catches my eye is that if it is a sentence piece tokenizer, self.tokenizer.legacy will always become True due to this line